armi/reactor/converters/axialExpansion/axialExpansionChanger.py

```diff
@@ -0,0 +1,396 @@
+        """
+        self.linked = AssemblyAxialLinkage(a)
+        self.expansionData = ExpansionData(
+            a, setFuel=setFuel, expandFromTinputToThot=expandFromTinputToThot
+        )
+        self._isTopDummyBlockPresent()
+
+    def applyColdHeightMassIncrease(self):
+        """
+        Increase component mass because they are declared at cold dims.
+
+        Notes
+        -----
+        A cold 1 cm tall component will have more mass that a component with the
+        same mass/length as a component with a hot height of 1 cm. This should be
+        called when the setting `inputHeightsConsideredHot` is used. This adjusts
+        the expansion factor applied during applyMaterialMassFracsToNumberDensities.
+        """
+        for c in self.linked.a.getComponents():
+            axialExpansionFactor = 1.0 + c.material.linearExpansionFactor(
+                c.temperatureInC, c.inputTemperatureInC
+            )
+            c.changeNDensByFactor(axialExpansionFactor)
+
+    def _isTopDummyBlockPresent(self):
+        """Determines if top most block of assembly is a dummy block.
+
+        Notes
+        -----
+        - If true, then axial expansion will be physical for all blocks.
+        - If false, the top most block in the assembly is artificially chopped
+          to preserve the assembly height. A runLog.Warning also issued.
+        """
+        blkLst = self.linked.a.getBlocks()
+        if not blkLst[-1].hasFlags(Flags.DUMMY):
+            runLog.warning(
+                f"No dummy block present at the top of {self.linked.a}! "
+                "Top most block will be artificially chopped "
+                "to preserve assembly height"
+            )
+            if self._detailedAxialExpansion:
+                msg = "Cannot run detailedAxialExpansion without a dummy block at the top of the assembly!"
+                runLog.error(msg)
+                raise RuntimeError(msg)
+
+    def axiallyExpandAssembly(self):
+        """Utilizes assembly linkage to do axial expansion.
+
+        .. impl:: Preserve the total height of an ARMI assembly, during expansion.
+            :id: I_ARMI_ASSEM_HEIGHT_PRES
+            :implements: R_ARMI_ASSEM_HEIGHT_PRES
+
+            The total height of an Assembly is preserved by not changing the ``ztop`` position
+            of the top-most Block in an Assembly. The ``zbottom`` of the top-most Block is
+            adjusted to match the Block immediately below it. The ``height`` of the
+            top-most Block is is then updated to reflect any expansion/contraction.
+        """
+        mesh = [0.0]
+        numOfBlocks = self.linked.a.countBlocksWithFlags()
+        runLog.debug(
+            "Printing component expansion information (growth percentage and 'target component')"
+            f"for each block in assembly {self.linked.a}."
+        )
+        for ib, b in enumerate(self.linked.a):
+            runLog.debug(msg=f"  Block {b}")
```

Comment:
Out of curiosity, do you think anyone but you will ever want to see these `debug` lines?

My concern is that they _look_ like something you added because they were useful to you developing axial expansion?